docs/cli/overview.md

```diff
@@ -0,0 +1,118 @@
+restore the dependencies, build the application and then run it. 
+
+```console
+dotnet new
+dotnet restore
+dotnet build --output /stuff
+dotnet /stuff/new.dll
+```
+
+### How does it work?
+As we saw in the short sample [above](#a-short-sample), there is a pattern in the way you use the CLI tools. Within that pattern, we can 
+identify three main pieces of each command:
+
+1. The driver ("dotnet")
+2. The command or, as we like to call it, the "verb"
+3. Command arguments
+
+Let's dig into more details on each of the above.
+
+### Driver
+The driver is named `dotnet`. It is the first part of what you invoke . The driver has two responsibilities:
+
+1. Executing IL code
+2. Executing the verb
+
+Which of the two things it does is dependent on what is specified on the command line. In the first case, you would 
+specify an IL assembly that `dotnet` would run similar to this: `dotnet /path/to/your.dll`. 
+
+In the second case, the driver would assume you want to invoke a command. This will start the CLI command execution 
+process. First, the driver will determine the version of the tooling that you want. You can specify the version in the 
+`global.json` file using the `sdkVersion` property. If that is not available, the driver will find the latest version
+of the tools that is installed on disk and will use that version. Once the version is determined, it will execute the 
+command. 
+
+### The "verb"
+The verb here is simply a command that performs an action. `dotnet build` will build your code. `dotnet publish` will publish 
+your code. The verb is implemented as a console application that is named per convention: `dotnet-{verb}`. All of the 
+logic is implemented in the console application that represents the verb. 
+
+### The arguments
+The arguments that you pass on the command line are the argument to the actual verb/command being invoked. 
+For example, when you type `dotnet publish --output publishedapp` the `--output` argument is passed to the 
+`publish` command. 
+
+## Type of application portability
+CLI enables applications to be portable in two main ways:
+
+1. Completely portable application that can run anywhere a shared component is installed
+2. Self-contained applications
+
+You can learn more about both of these in the [application types overview](app-types.md) topic. 
+
+## Migration from DNX
+If you used DNX in the RC1 time frame of .NET Core, you may be wondering what happened to it and how do these new tools
+relate to the DNX tools. In short, DNX has been discontinued and is completely replaced by the CLI tooling. In RC2 and 
+beyond, you will not be using DNX anymore. If you have existing projects or are just wondering how the commands map, you 
+can use the [DNX to CLI migration document](dnx-migration.md) to get all of the details. 
+
+## Extensibility
+Of course, not every tool that you could use in your workflow will be a part of the core CLI tools. However, we do have 
```

Comment:
Try to avoid "we" wherever possible.  How about ".NET Core has an extensibility model"...